RelationalAI/CommonPolicies.cs

```diff
@@ -41,41 +41,47 @@ static CommonPolicies()
         {
             return policyBuilder
-                .WaitAndRetryForeverAsync(retryAttempt => GetBoundedRetryDelay(retryAttempt, delayThreshold))
+                .WaitAndRetryForeverAsync(retryAttempt => GetBoundedRetryDelay(retryAttempt, startTime, overheadRate, delayThreshold))
                 .AddRequestErrorResilience();
         }
 
         /// <summary>
         /// Creates a policy that can be used to produce synchronous API for async calls which may be taking less than 15 min
         /// to complete, otherwise throws TimeoutRejectedException.
-        /// Uses exponential back-off starting with a 2 seconds delay up to 15 seconds and then retrying every 15 seconds.
+        /// Uses exponential back-off with an overhead % of the time the transaction has been running so far up to 15 seconds and then retrying every 15 seconds.
         /// Retries on HTTP request sending errors.
         /// </summary>
         /// <typeparam name="T">Result type of an operation.</typeparam>
         /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="startTime">transaction startTime epoch milliseconds</param>
+        /// <param name="overheadRate">the overhead % to add through polling</param>
         /// <returns>Resulting policy instance.</returns>
-        public static AsyncPolicy<T> Retry15Min<T>(this PolicyBuilder<T> policyBuilder)
+        public static AsyncPolicy<T> Retry15Min<T>(this PolicyBuilder<T> policyBuilder, long startTime, double overheadRate = 0.01)
         {
-            return policyBuilder.AddBoundedRetryPolicy(15, 15 * 60);
+            return policyBuilder.AddBoundedRetryPolicy(startTime, overheadRate, 15, 15 * 60);
         }
 
         /// <summary>
         /// Creates a policy that can be used to produce synchronous API for async calls which may be taking less than 30 min
         /// to complete, otherwise throws TimeoutRejectedException.
-        /// Uses exponential back-off starting with a 2 seconds delay up to 15 seconds and then retrying every 15 seconds.
+        /// Uses exponential back-off with an overhead % of the time the transaction has been running so far up to 15 seconds and then retrying every 15 seconds.
         /// Retries on HTTP request sending errors.
         /// </summary>
         /// <typeparam name="T">Result type of an operation.</typeparam>
         /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="startTime">transaction startTime epoch milliseconds</param>
+        /// <param name="overheadRate">the overhead % to add through polling</param>
         /// <returns>Resulting policy instance.</returns>
-        public static AsyncPolicy<T> Retry30Min<T>(this PolicyBuilder<T> policyBuilder)
+        public static AsyncPolicy<T> Retry30Min<T>(this PolicyBuilder<T> policyBuilder, long startTime, double overheadRate = 0.01)
```

Comment:
Ok will clean those up

Reply:
I might miss some details around async etc. (so some of this might be mandatory) but it's not straight forward to follow the logic. In the go sdk it is very straightforward to see what is happening (e.g. what's the % backoff per retry).